src/ci/docker/dist-x86_64-musl/Dockerfile

```diff
@@ -39,8 +38,17 @@ ENV RUST_CONFIGURE_ARGS \
 # way to produce "super compatible" binaries.
 #
 # See: https://github.com/rust-lang/rust/issues/34978
-ENV CFLAGS_x86_64_unknown_linux_musl=-Wa,-mrelax-relocations=no
+#ENV CFLAGS_x86_64_unknown_linux_musl=-Wa,-mrelax-relocations=no
```

Comment:
Do you know if this is something we still need to fix? Or does using a different toolchain fix this?

Reply:
It was disabled by @strfry during his tries: https://github.com/rust-lang/rust/pull/58575/commits/1624dc08126b054db906a6a2e2d8ee071803a758
I don't know reasoning for it. Last time I checked my checkout it was uncommented but I realized just now I was on a different branch.

I'll enable it an retest if produced compiler still works.